indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -66,6 +66,22 @@
   public String getId();
 
   /**
+   * Returns priority of this task. Task priority is used for acquiring lock on an interval for a datasource.
+   * Task with higher priority override the task with lower priority for overlapping interval for a datasource if ran concurrently.
+   *
+   * Legacy tasks or tasks with "0" priority will have the respective default priorities as per the task type
+   *  - Default priorities for task
+   *    - Realtime Index Task - 100
+   *    - Hadoop/Index Task - 75
+   *    - Merge/Append Task - 50
+   *    - Other Tasks - 25
+   * Higher the number, higher the priority. Default priority can be overridden by specifying it manually in the task json.
+   *
+   * @return task priority
+   * */
+  public int getPriority();
```

Comment:
`getLockPriority`? IIRC we thought there might be different kinds of priorities in the future with different semantics.